test/python/primitives/test_backend_sampler.py

```diff
@@ -0,0 +1,565 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for BackendSampler."""
+
+import unittest
+from test import combine
+
+import numpy as np
+from ddt import ddt
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.exceptions import QiskitError
+from qiskit.primitives import BackendSampler, SamplerResult
+from qiskit.providers import JobStatus, JobV1
+from qiskit.test import QiskitTestCase
+from qiskit.providers.fake_provider import FakeNairobi, FakeNairobiV2
+
+BACKENDS = [FakeNairobi(), FakeNairobiV2()]
+
+
+@ddt
+class TestBackendSampler(QiskitTestCase):
+    """Test BackendSampler"""
+
+    def setUp(self):
+        super().setUp()
+        hadamard = QuantumCircuit(1, 1)
+        hadamard.h(0)
+        hadamard.measure(0, 0)
+        bell = QuantumCircuit(2)
+        bell.h(0)
+        bell.cx(0, 1)
+        bell.measure_all()
+        self._circuit = [hadamard, bell]
+        self._target = [
+            {0: 0.5, 1: 0.5},
+            {0: 0.5, 3: 0.5, 1: 0, 2: 0},
+        ]
+        self._pqc = RealAmplitudes(num_qubits=2, reps=2)
+        self._pqc.measure_all()
+        self._pqc2 = RealAmplitudes(num_qubits=2, reps=3)
+        self._pqc2.measure_all()
+        self._pqc_params = [[0.0] * 6, [1.0] * 6]
+        self._pqc_target = [{0: 1}, {0: 0.0148, 1: 0.3449, 2: 0.0531, 3: 0.5872}]
+        self._theta = [
+            [0, 1, 1, 2, 3, 5],
+            [1, 2, 3, 4, 5, 6],
+            [0, 1, 2, 3, 4, 5, 6, 7],
+        ]
+
+    def _generate_circuits_target(self, indices):
+        if isinstance(indices, list):
+            circuits = [self._circuit[j] for j in indices]
+            target = [self._target[j] for j in indices]
+        else:
+            raise ValueError(f"invalid index {indices}")
```

Comment:
Seems a little odd to have this defensive code in test helpers.

Reply:
yeah it does, I just copied the tests from the existing reference sampler implementation tests and updated it to use `BackendSampler` instead of `Sampler`. I will remove this though since it probably isn't needed